scripts/sort_sparse_matrix.py

```diff
@@ -0,0 +1,65 @@
+"""Python Function to sort sparse expression matrix files
+Arguments:
+   matrix_file  sparse matrix file to sort
+   -o, --sorted_matrix_file  optional output path of sorted file, else "gene_sorted-" + matrix_file
+"""
+
+# Imports
+import pandas as pd
+import sys
+import argparse
+
+
+def sort_sparse_matrix(matrix_file, sorted_matrix_file=None):
+	"""Sort the sparse matrix file by gene, barcode
+		Arguments:
+		matrix_file- sparse matrix file to sort
+		sorted_matrix_file- optional output path of sorted file
+		Outputs-
+		output_name- gene, barcode sorted sparse matrix file, name is either sorted_matrix_file if provided, or "gene_sorted-" + matrix_file
+	"""
+	if sorted_matrix_file:
+		# if the sorted path name is provided, use it as the output name
+		output_name = sorted_matrix_file
+	else:
+		# otherwise if the sorted path name is not provided, add "gene_sorted-" to the original path name
+		output_name = "gene_sorted-" + matrix_file
+	# read sparse matrix
+	print("Reading Sparse Matrix")
+	headers = []
+	with open(matrix_file) as matrix:
+		line = next(matrix)
+		i = 1
+		while line.startswith("%"):
+			headers = headers + [line]
+			line = next(matrix)
+			i += 1
+		headers = headers + [line]
+		df = pd.read_table(matrix, sep=" ", header=i, names=['genes', 'barcodes', 'expr'])
```

Comment:
This is better - it handles the header lines. However, I'm worried about calling pd.read_table here multiple times. If that matrix is large, that could bog down, or at the least increase execution time. It may be simpler to read the file just as a normal text file and keep track of how many lines we're disregarding (via i, or some other variable) and then only call pd.read_table once we know how many lines to ignore.